torch/_inductor/kernel/flex_decoding.py

```diff
@@ -546,8 +546,8 @@ def create_flex_decoding_kernel(*args, **kwargs):
     # See [Note] Handle fully masked out rows:
     # g_M Is the global max among split kv blocks.
     masked_rows = lowerings[aten.eq](g_M, -float("inf"))
-    g_M = lowerings[aten.where](masked_rows, 0.0, g_M)
     adj_M = lowerings[aten.sub](buf_M, g_M)
+    adj_M = lowerings[aten.where](masked_rows, 0, adj_M)
```

Comment:
The previous order was actually incorrect for the semantics where `logsumexp` of a masked out row is `-inf`. Specifically, the previous masking was done because we're computing the "difference in max values" so we can rescale it appropriately, and if `g_M` and `buf_M` were both `-inf` this would cause nans (i.e. `-inf - (-inf) => nan`).

However, we then compute `log(L) + M`. If `M` was masked to 0 here, then the end result would be 0, which is not the `logsumexp` value we want.